src/praisonai-mcp/praisonai_mcp/mcp_server/agent_adapter.py

```
@@ -0,0 +1,171 @@
+        user_id: Optional[str] = None,
+    ) -> Dict[str, Any]:
+        # Mint a fresh id when omitted — never fall back to a shared session.
+        sid = session_id or uuid.uuid4().hex
+        history = store.load(sid)
+
+        # Isolate per-call so concurrent sessions never share mutable state.
+        turn_agent = _isolated_agent(agent)
+        try:
+            turn_agent.chat_history = list(history)
+        except Exception:
+            pass
```

Ruff (0.16.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)